parallel_test.go

```diff
@@ -0,0 +1,45 @@
+package autogold_test
+
+import (
+	"testing"
+
+	"github.com/hexops/autogold"
+)
+
+func TestWant_parallel1(t *testing.T) {
+	testParallel(t, "1")
+}
+
+func TestWant_parallel2(t *testing.T) {
+	testParallel(t, "2")
+}
+
+func TestWant_parallel3(t *testing.T) {
+	testParallel(t, "3")
+}
+
+func TestWant_parallel4(t *testing.T) {
+	testParallel(t, "4")
+}
+
+func testParallel(t *testing.T, prefix string) {
+	t.Parallel()
+
+	testTable := []string{
+		prefix + "-first",
+		prefix + "-second",
+		prefix + "-third",
+		prefix + "-fourth",
+		prefix + "-fifth",
+	}
+
+	for _, name := range testTable {
+		name := name
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			autogold.Equal(t, name)
```

Comment:
This adds quite a substantial number of small files to the repo.

Maybe you'd be open to adding a private/internal option that we can pass to `Equal` here like `autogold.Equal(t, name, testNoWrite())` and, when that option is present, have `Equal` not write the actual golden files? (since they're not really useful here)

Reply:
That sounds reasonable, I’ll have a look into what’s required. It’s likely the number of tests and files could also be reduced while still reproducing the issue

Reply:
I was able to cut the number of tests/files down to 2 top-level tests, each with 2 subtests, making for 4 files total, while still reliably reproducing the issue.